### ISSUE TITLE
Invalid characters on file name

### ISSUE THREAD

**Issue by tzarebczan** [lbryio/lbry-sdk#2084]
## The Issue
Recently ran into a file that was uploaded by a mac user, where I assume a ? is a valid character in a file name. But on windows, the file will fail to save because it's invalid. Should we not allow this to begin with or strip it out automatically on download?

```2019-05-06 08:59:45,048 INFO     lbrynet.stream.managed_stream:333: save file for lbry://whycantbitcoinbeshutdownorregulated#bdb94799b6bafd444985f7ec8e8ea85af9982fdc (sd hash 23d1e1...) -> c:\testing\lbry036\downloads\Why can't Bitcoin be shut down or regulated?.mov
2019-05-06 08:59:45,048 ERROR    lbrynet.stream.managed_stream:359: unexpected error encountered writing file for stream 23d1e1afe89821db541513fa786d5bd69d777bae67d49c0746ec5277db764699097cee8f91b3fcce0444dec705163bc3
Traceback (most recent call last):
  File "c:\users\thoma\documents\lbry\lbrynet\stream\managed_stream.py", line 338, in _save_file
    with open(output_path, 'wb') as file_write_handle:
OSError: [Errno 22] Invalid argument: "c:\\testing\\lbry036\\downloads\\Why can't Bitcoin be shut down or regulated?.mov"
```
## System Configuration

 

- LBRY Daemon version:
- LBRY App version:
- LBRY Installation ID:
- Operating system:

### Acceptance Criteria
1.
2.
3.

### Definition of Done
- [ ]  Tested against acceptance criteria
- [ ] Tested against the assumptions of user story
- [ ] The project builds without errors
- [ ] Unit tests are written and passing
- [ ] Tests on devices/browsers listed in the issue have passed
- [ ] QA performed & issues resolved
- [ ] Refactoring completed
- [ ] Any configuration or build changes documented
- [ ] Documentation updated
- [ ] Peer Code Review performed

**Comment by mirgee** [comment]
@tzarebczan I think that disallowing it on upload is bad UX and unncessary. It is easy to forget that we sanitize something, so I am for implementing sanitization during both upload and download. If I was to choose one, I would go with the upload - since AFAIK the client side can be bypassed, we shouldn't broadcast streams with potentially invalid metadata in the first place. Is there some concensus on this?

**Comment by mirgee** [comment]
@tzarebczan On the other hand, the [spec](https://lbry.tech/spec#metadata-validation) reads:

> [...] Clients should not trust the metadata they read from the blockchain. Each client is responsible for correctly encoding and decoding the metadata, and for validating its structure and signatures. This allows evolution of the metadata definition without changes to blockchain consensus rules.

It says nothing about sanitization, but it may as well apply apply to it by extension.

**Comment by lyoshenka** [comment]
fixed by #2476
